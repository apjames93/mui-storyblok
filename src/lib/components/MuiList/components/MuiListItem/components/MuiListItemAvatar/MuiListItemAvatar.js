import React from 'react';
import PropTypes from 'prop-types';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import Avatar from '@material-ui/core/Avatar';
import { validComponentsRequired } from '../../../../../../utils/customProps';
import Storyblok from '../../../../../../utils/Storyblok';
import { renderComponents } from '../../../../../../utils/customComponents';
import MuiIcon from '../../../../../MuiIcon/MuiIcon';

/**
 * MuiListItemAvatar
 */

export const MuiListItemAvatar = ({
  content,
  rootClass,
}) => {
  const components = {
    MuiIcon,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <ListItemAvatar className={styles.root}>
      <Avatar>
        {content.map((component, key) => renderComponents(components, component, key))}
      </Avatar>
    </ListItemAvatar>
  );
};

export default MuiListItemAvatar;

MuiListItemAvatar.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Override or extend the styles applied to the component
   * */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /** MuiIcon Allowed maximum: 1 */
  content(props, propName, componentName) {
    return validComponentsRequired(props, propName, componentName, ['MuiIcon'], 1);
  },
};

MuiListItemAvatar.defaultProps = {
  rootClass: [],
  content: [],
};
