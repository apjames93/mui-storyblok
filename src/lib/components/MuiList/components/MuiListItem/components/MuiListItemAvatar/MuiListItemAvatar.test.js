import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiListItemAvatar from './MuiListItemAvatar';

function setup() {
  const props = {
    content: [{
      _editable: '<!--#storyblok#{"name": "MuiIcon", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiIcon',
      iconName: 'android',
    }],
  };
  const comp = shallow(<MuiListItemAvatar {...props} />);
  return { comp, props };
}

describe('<MuiListItemAvatar />', () => {
  it('renders MuiListItemAvatar', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<MuiListItemAvatar {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
