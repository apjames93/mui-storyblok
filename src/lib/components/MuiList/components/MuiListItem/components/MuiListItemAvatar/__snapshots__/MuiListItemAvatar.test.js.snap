// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiListItemAvatar /> snapshot 1`] = `
<div
  className="MuiListItemAvatar-root makeStyles-root-3"
>
  <div
    className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
  >
    <span
      aria-hidden={true}
      className="material-icons MuiIcon-root makeStyles-root-4 MuiIcon-colorSecondary"
    >
      android
    </span>
  </div>
</div>
`;
